modules/engage-paella-player/src/main/paella-opencast/plugins/es.upv.paella.opencast.loader/01_prerequisites.js

```diff
@@ -0,0 +1,143 @@
+
+paella.opencast = new (Class ({
+	_me: undefined,
+	_episode: undefined,
+	_series: undefined,
+	_acl: undefined,
+	
+	getUserInfo:function() {	
+		var self = this;	
+		return new Promise((resolve, reject)=>{
+			if (self._me) {
+				resolve(self._me);
+			}
+			else {
+				base.ajax.get({url:'/info/me.json'},
+					function(data,contentType,code) {
+						self._me = data;
+						resolve(data);
+					},
+					function(data,contentType,code) { reject(); }
+				);
+			}				
+		});	
+	},
+	
+	getEpisode: function() {
+		var self = this;		
+		return new Promise((resolve, reject)=>{
+			if (self._episode) {
+				resolve(self._episode);
+			}
+			else {
+				var episodeId = paella.utils.parameters.get('id');
+				base.ajax.get({url:'/search/episode.json', params:{'id': episodeId}},
+					function(data, contentType, code) {
+						if (data['search-results'].result) {
+							self._episode = data['search-results'].result;
+							resolve(self._episode);		
+						}
+						else {
+							reject();						
+						}
+					},
+					function(data, contentType, code) {
+						reject();
+					}
+				);
+			}		
+		});		
+	},
+	
+	
+	getSeries: function() {
```

Comment:
This is just a comment, I can make it a pull towards the UPV paella-opencast if it sounds reasonable.

Our site experienced occasional problems when this getSeries calls the series-remote endpoint on engage (presentation) node (using OCv1x+). The series-remote calls back to the series-impl on admin node. This makes paella dependent on the admin node being up and the connection between admin and engage being up. Any slowness or other other issues between engage & admin directly affect the loading of the player for the client at this point. To avoid the admin dependency, we changed the getSeries to get series data from the Search service's search/series REST endpoint. It contains the same sort of series metadata and the search-impl is native on engage.

Reference:
https://github.com/harvard-dce/paella-matterhorn/blob/master/paella-opencast/plugins/es.upv.paella.opencast.loader/01_prerequisites.js#L80

Reply:
and it makes paella need to use one less service from Opencast.

Reply:
@karendolan It sounds OK to me.